app/src/main/java/ca/klapstein/baudit/Data/PhoneNumber.java

```diff
@@ -2,24 +2,34 @@
 
 import android.support.annotation.NonNull;
 
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 public class PhoneNumber {
     private static final String TAG = "PhoneNumber";
+    private static final String VALID_TEN_DIGIT_PHONE_NUMBER
+        = "\\(?(\\d{3})\\)?(?: ?|-?)(\\d{3})(?: ?|-?)(\\d{4})";
 
     @NonNull
     private String phoneNumber;
 
-    public PhoneNumber(@NonNull String phoneNumber) {
-        // TODO: add validation of phone number string
-        this.phoneNumber = phoneNumber;
+    public PhoneNumber(@NonNull String phoneNumber) throws IllegalArgumentException {
+        this.setPhoneNumber(phoneNumber);
     }
 
+    @NonNull
     public String getPhoneNumber() {
-        return phoneNumber;
+        return this.phoneNumber;
     }
 
-    public void setPhoneNumber(String phoneNumber) {
-        // TODO: add validation of phone number string
-        this.phoneNumber = phoneNumber;
+    public void setPhoneNumber(String phoneNumber) throws IllegalArgumentException {
+        Pattern p = Pattern.compile(VALID_TEN_DIGIT_PHONE_NUMBER);
```

Comment:
you should separate the validation of a phoneNumber string into a separate method.

e.g.
```java
public boolean isValid(String phoneNumber)
```